src/turn.c

```diff
@@ -112,13 +112,13 @@ static int find_ordered_transaction_id_rec(turn_entry_t *const ordered_transacti
                                            const uint8_t *transaction_id, int begin, int end) {
 	int d = end - begin;
 	if (d <= 0)
-		return begin;
+		return -1;
```

Comment:
Actually, the function is also used to get the index where the element should be for insertion, so the index should still be returned if there is no element..

Reply:
You mean in `set_transaction_id` right?

What about the other uses:
* `remove_ordered_transaction_id`
* `update_timestamp`
* `turn_find_transaction_id`

Those functions check whether the returned index equals `transaction_ids_count` to determine whether the id exists.
Will this still work when we return the position where the element would potentially be inserted?

Reply:
Yes that's right, actually these functions check if the index is strictly less than `transaction_ids_count` _and_ that the transaction id at index is the expected one.

Reply:
Ahh ok. Now I see it...  didn't look into the callers of `remove_ordered_transaction_id` before :facepalm: 
Thanks for the clarification.

I have comitted the suggested change.

Reply:
Excellent, thank you :+1: